webhooks/main.py

```diff
@@ -40,44 +54,58 @@ async def wallet_root(
     return {f"{wallet_id}": value}
 
 
-@app.post("/topic/{topic}")
+# 'origin' helps to distinguish where a hook is from
+# eg the admin, tenant or OP agent respectively
+@app.post("/{origin}/topic/{topic}")
 @inject
 async def topic_root(
-    topic, request: Request, service: Service = Depends(Provide[Container.service])
+    topic,
+    origin,
+    request: Request,
+    service: Service = Depends(Provide[Container.service]),
 ):
-    payload = pformat(await request.json())
-    await service.add_topic_entry(str(topic), str(payload))
+    payload = await request.json()
+    wallet_id = {"wallet_id": request.headers["x-wallet-id"]}
+    payload.update(wallet_id)
+    origin_id = {"origin": origin}
+    payload.update(origin_id)
+    payload = pformat(payload)
```

Comment:
Instead of doing `str(pformat(payload))` can't we do `json.dumps(payload)` and encode it to bytes? Seems less hacky to me